### ISSUE TITLE
S1481 - False positive when using destructuring

### ISSUE THREAD

**Issue by rjgotten** [SonarSource/sonar-dotnet#2126]
### Description

`S1481 - Remove this unused '<name>' local variable` is raised incorrectly when the variable in question is used in a foreach loop that employs destructuring assignment.

### Repro steps

Run analysis on code which uses destructuring, similar to the following:

```csharp
public void SomeMethod()
{
  var items = new SomeDestructurableType[] { ... };

  foreach(var (first, second) in items)
  {
    ...
  }
}
```

### Expected behavior

S1481 should not be raised.

### Actual behavior

`S1481 - Remove this unused 'items' local variable` is raised.

### Known workarounds

Not using the destructuring language feature.

### Related information

(No idea how to find version information for which SonarC# version is being used.)

**Comment by valhristov** [comment]
Hi @rjgotten, I tried with the following code and cannot reproduce the problem with the latest version of SonarC#:
```
namespace ClassLibrary1
{
    public class Class1
    {
        public void SomeMethod()
        {
            var items = new SomeDestructurableType[] { }; // S1481 is not raising here

            var items2 = new SomeDestructurableType[] { }; // S1481 is raising here

            foreach (var (first, second) in items)
            {
                // Do something
            }
        }
    }

    internal class SomeDestructurableType
    {
        public void Deconstruct(out string key, out string value)
        {
            key = value = "";
        }
    }
}
``` 

To see what version of our analyzers are in use in your project:
- if you analyze your project with Scanner for MSBuild, you could open the following URL with your browser `http://localhost:9000/api/plugins/installed` (you will need to replace the part before /api with the actual URL of your SonarQube server) and look for `SonarC#`. If the version is below 7.8 you will need to update the plugin (or ask your administrator to update it for you).
- if you use SonarLint for Visual Studio, open Visual Studio, click the menu `Help -> About`, find SonarLint for Visual Studio. If the version is below 4.4, you will need to update the extension.
- if you use the nuget analyzers, open your project in Visual Studio, right click on the project, choose `Manage Nuget Packages`, find SonarAnalyzer.CSharp. If the version is below 7.8, you will need to update the package.

**Comment by valhristov** [comment]
Closing this ticket because of lack of activity. Please, feel free to open it again.
